packages/ui/src/components/Select/SSelectDropdown.vue

```diff
@@ -24,30 +27,78 @@ const optionGroups: ComputedRef<SelectOptionGroup[]> = computed(() => {
   return options
 })
 
+const isSearching = eagerComputed(() => props.search && api.searchQuery)
+
+const shownOptionGroups: ComputedRef<SelectOptionGroup[]> = computed(() => {
+  if (!api.searchQuery) {
+    return optionGroups.value
+  }
+
+  const escapedQuery = new RegExp(escapeStringRegexp(api.searchQuery), 'i')
+
+  return optionGroups.value.map((x) => ({ ...x, items: x.items.filter((x) => escapedQuery.test(x.label)) }))
```

Comment:
I'd move `new RegExp` into a separate `computed`, so it will depend purely on `api.searchQuery` and will not recompute each time with no need.

Reply:
fixed